docs/literate/src/files/structured_mesh_mapping.jl

```diff
@@ -0,0 +1,197 @@
+# More precisely, this tutorial is about the curved mesh type [`StructuredMesh`](@ref) which supports
+# fully curved meshes created for instance by an input mapping.
+
+# # Creating a curved mesh
+# There are two basic options to define a curved [`StructuredMesh`](@ref) in Trixi. You can
+# implement functions for the boundaries, or alternatively, set up the complete transformation mapping.
+# We now present a short example each.
+
+# ## Mesh defined by boundary functions
+# Both examples are based on a semdiscretization of the 2D compressible Euler equations.
+
+using OrdinaryDiffEq
+using Trixi
+
+equations = CompressibleEulerEquations2D(1.4)
+
+# We start with a pressure perturbation at `(xs, 0.0)` as initial condtition.
+function initial_condition_pressure_perturbation(x, t, equations::CompressibleEulerEquations2D)
+  xs = 1.5 # location of the initial disturbance on the x axis
+  w = 1/8 # half width
+  p = exp(-log(2) * ((x[1]-xs)^2 + x[2]^2)/w^2) + 1.0
+  v1 = 0.0
+  v2 = 0.0
+  rho = 1.0
+
+  return prim2cons(SVector(rho, v1, v2, p), equations)
+end
+initial_condition = initial_condition_pressure_perturbation
+
+# Initialize every boundary as a [`boundary_condition_slip_wall`](@ref).
+boundary_conditions = boundary_condition_slip_wall
+
+# The approximation setup is a split-form DG method and `polydeg=4`. We are using the two
+# fluxes [`flux_ranocha`](@ref) and [`flux_lax_friedrichs`](@ref).
+solver = DGSEM(polydeg=4, surface_flux=flux_lax_friedrichs,
+               volume_integral=VolumeIntegralFluxDifferencing(flux_ranocha))
+
+# We want to define a circular cylinder as physical domain. It contains of an inner semicircle with
+# radius `r0` and an outer semicircle of radius `r1`.
+
+# TODO: add picture
+
+#src # \documentclass{standalone}
+#src # \usepackage{tikz}
+#src # \begin{document}
+#src #   \begin{tikzpicture}
+#src #     % Circular cylinder
+#src #     \draw[->] (0.5,0) -- node[above] {$\textbf{f1}$} (5,0);
+#src #     \draw[<-] (-5,0)  -- node[above] {$\textbf{f2}$} (-0.5,0);
+#src #     \draw[->] (0.5,0) arc (0:180:0.5); \node at (0, 0.8) {$\textbf{f3}$};
+#src #     \draw[->] (5,0)   arc (0:180:5);   \node at (0, 4.7) {$\textbf{f4}$};
+#src #     \draw[dashed] (-5, -0.5)   -- node[below] {$\textbf{r}_1$} (0, -0.5);
+#src #     \draw[dashed] (-0.5, -0.7) -- node[below] {$\textbf{r}_0$} (0, -0.7);
+#src #     % Arrow
+#src #     \draw[line width=0.1cm, ->, bend angle=30, bend left] (4, 2) to (8,3);
+#src #     % Right Square
+#src #     \draw[->] (7,0)  -- node[below] {$\textbf{f1}$} (12,0);
+#src #     \draw[->] (7,5)  -- node[above] {$\textbf{f2}$} (12,5);
+#src #     \draw[->] (7,0)  -- node[left]  {$\textbf{f3}$} (7,5);
+#src #     \draw[->] (12,0) -- node[right] {$\textbf{f4}$} (12,5);
+#src #     % Pressure perturbation
+#src #     \draw[fill] (1.6, 0) arc (0:180:0.1);
+#src #     \draw (1.7, 0) arc (0:180:0.2);
+#src #     \draw (1.8, 0) arc (0:180:0.3);
+#src #   \end{tikzpicture}
+#src # \end{document}
```

Comment:
Nice! I think it would be nice to ad this source code to the document (without showing it in the tutorial) and paste the output here. Then, you can include the picture from GitHub as usual in the tutorial.